lib/src/main/ktx/com/fabernovel/statefullayout/StatefulLayout.kt

```diff
@@ -106,3 +109,62 @@ fun StatefulLayout.showLoading(): State = showState(R.id.stateLoading)
  * @return the content state
  */
 fun StatefulLayout.showContent(): State = showState(R.id.stateContent)
+
+/**
+ * Sets a minimal time to wait after each successful state change triggered by [requestStateChange].
+ *
+ * All requests triggered from [requestStateChange] will be handled in respect of this minimal
+ * time, sometimes dropping some requests when many (backpressure).
+ *
+ * Default value for [timeToWaitInMillis] is [DEFAULT_REGULATOR_PACE]
+ *
+ * @see [StateChangeRegulator] for details about regulation rules.
+ */
+@FlowPreview
+@ExperimentalCoroutinesApi
+fun StatefulLayout.setMinimalTimeToWaitBetweenStateChanges(timeToWaitInMillis: Long) {
+    stateChangeRegulator.minimalTimeBetweenRequestsMillis = timeToWaitInMillis
+}
+
+/**
+ * Posts a request to change state.
+ *
+ * Not all requests will lead to a state change since they will all go through regulation defined
+ * by [setMinimalTimeToWaitBetweenStateChanges]
+ *
+ * @see StateChangeRegulator
+ */
+@FlowPreview
+@ExperimentalCoroutinesApi
+fun StatefulLayout.requestStateChange(
+    @IdRes stateId: Int,
+    showTransitions: Boolean = areTransitionsEnabled
+) {
+    viewCoroutineScope.launch {
+        val request = StateChangeRequest(stateId, showTransitions)
+        stateChangeRegulator.requestStateChange(request)
+    }
+}
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+private val StatefulLayout.stateChangeRegulator: StateChangeRegulator
+    get() {
+        var regulator = getTag(R.id.stateChangeRegulator) as StateChangeRegulator?
```

Comment:
To avoid concurrency issue, we should throw an exception if the getter is not called from the ui thread.